src/NaCl.Core/Base/SalsaIntrinsics/Salsa512.cs

```diff
@@ -0,0 +1,243 @@
+﻿#if INTRINSICS
```

Comment:
I wonder if we do need the preprocessor directive since we are checking if the hardware supports it